src/client/ua_client_internal.h

```diff
@@ -111,16 +111,19 @@ void UA_Client_AsyncService_cancel(UA_Client *client, AsyncServiceCall *ac,
 
 void UA_Client_AsyncService_removeAll(UA_Client *client, UA_StatusCode statusCode);
 
+typedef void (*CustomCallbackDataDeleter)(void *);
```

Comment:
Why do we need this guy?
If there is a timeout, we can just set a statuscode in the responseheader and call the "normal" callback.

Reply:
My motivation was to have the delete function in the custom callback to be able to free the memory if the handler isnt called. It looks like that the handler is always called (remove UA_Client_AsyncService_cancel) so i can remove it from the CustomCallback.

Reply:
I have replaced the deleter with a flag to distinguish between "async customcallbacks) on the heap and (sync customcallbacks) on the stack.